internal/trie/node/node.go

```diff
@@ -3,24 +3,87 @@
 
 package node
 
-import "github.com/qdm12/gotree"
+import (
+	"fmt"
+
+	"github.com/qdm12/gotree"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
 
 // Node is a node in the trie and can be a leaf or a branch.
-type Node interface {
-	Encode(buffer Buffer) (err error) // TODO change to io.Writer
-	EncodeAndHash(isRoot bool) (encoding []byte, hash []byte, err error)
-	ScaleEncodeHash() (encoding []byte, err error)
-	IsDirty() bool
-	SetDirty(dirty bool)
-	SetKey(key []byte)
-	String() string
-	SetEncodingAndHash(encoding []byte, hash []byte)
-	GetHash() (hash []byte)
-	GetKey() (key []byte)
-	GetValue() (value []byte)
-	GetGeneration() (generation uint64)
-	SetGeneration(generation uint64)
-	Copy(settings CopySettings) (cpy Node)
-	Type() Type
-	StringNode() (stringNode *gotree.Node)
+type Node struct {
+	// Key is the partial key bytes in nibbles (0 to f in hexadecimal)
+	Key   []byte
+	Value []byte
+	// Generation is incremented on every trie Snapshot() call.
+	// Each node also contain a certain Generation number,
+	// which is updated to match the trie Generation once they are
+	// inserted, moved or iterated over.
+	Generation uint64
+	// Children is a slice of length 16 for branches.
+	// It is left to nil for leaves to reduce memory usage.
+	Children []*Node
```

Comment:
Looks like this was sized to `Children [16]Node` earlier. why this change to unsized?

Reply:
So it can be nil for leaves and not use 16x8 bytes per leaf of space. It's always made to a slice of 16 children for branches though.